python/cudf/cudf/_libxx/column.pyx

```diff
@@ -293,15 +293,18 @@ cdef class Column:
         return self._view(libcudf_types.UNKNOWN_NULL_COUNT).null_count()
 
     cdef mutable_column_view mutable_view(self) except *:
+        col_offset = self.offset
```

Comment:
You need to similarly handle the size in the case someone slices off from the end, i.e. `col[:2]`